### PR TITLE
Splitting functionality from jenkins_build to script libraries so they can be reused

### PR DESCRIPTION
**Pull request by alexgg**
Specifically there are:

* balena-build.sh: This script can be used standalone to call a containerized barys build. Helpful for customers that don't want to setup an OS and to keep a the same shared state between different machines.
* The `balena-deploy.inc` file contains a `balena_deploy_hostapp` function that deploys a hostapp.

### CODE REVIEW

File: automation/balena-lib.inc [hunk 1]
```diff
@@ -28,3 +46,97 @@ balena_lib_docker_pull_helper_image() {
...
+	BALENARC_BALENA_URL=${_api_env} balena login --token "${_token}"
+}
+
+# Login to dockerhub
+balena_lib_dockerhub_login() {
+    local _user="${1}"
+    local _pwd="${2}"
+
+    if [ -z "${_user}" ]; then
+        if [ -n "${JOB_NAME}" ]; then
```
Comment (klutchell) [inline automation/balena-lib.inc:128]: I can't find where `JOB_NAME` is actually used. Is it a legacy variable we should clean up?

File: automation/jenkins_build-containers.sh [hunk 2]
```diff
@@ -8,16 +8,9 @@ SCRIPTPATH=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
...
+source "${SCRIPTPATH}/balena-lib.inc"
 
-if [ -z "${JOB_NAME}" ]; then
-    echo "[ERROR] No job name specified."
-    exit 1
-fi
-
-echo "Login to docker as ${DOCKERHUB_USER}"
-docker login -u "${DOCKERHUB_USER}" -p "${DOCKERHUB_PWD}"
+balena_lib_dockerhub_login
```
Comment (klutchell) [inline automation/jenkins_build-containers.sh:13]: I guess in this case we are expecting DOCKERHUB_USER and DOCKERHUB_PWD to be set in the env so we aren't passing args?

File: automation/balena-lib.inc (hunk 1 above)
Comment (alexgg) [reply automation/balena-lib.inc:128]: @klutchell it's set by Jenkins automatically. Here it is used to distinguish whether we are called as part of a Jenkins build so the credentials are expected to be in the environment, or not, in which case the login will ask the user for credentials (this is for local builds)

File: automation/jenkins_build-containers.sh (hunk 2 above)
Comment (alexgg) [reply automation/jenkins_build-containers.sh:13]: @klutchell if it's a Jenkins build then yes, if not "docker login" will either used cached credentials or prompt the user.

File: automation/balena-lib.inc (hunk 1 above)
Comment (klutchell) [reply automation/balena-lib.inc:128]: Ah, of course, I forgot that we use JOB_NAME to determine if it's Jenkins.

**Comment by alexgg** [comment]
@balena-ci rebase

**Comment by alexgg** [comment]
Tested with a raspberrypi4-64 build & staging deployment and resolved all the review comments.

**Comment by alexgg** [comment]
@balena-ci I self-certify!

**Pull request merged**